### PR TITLE
3720: Panic if there is an overflow

### PR DESCRIPTION
**Pull request by real-felix**
Computing the sum of stakes panics, if there is an overflow.

### CODE REVIEW

**Review by alsrdn — APPROVED** [review]

File: node/src/components/consensus/protocols/common.rs [hunk 1]
```diff
@@ -17,11 +17,14 @@ pub(crate) fn validators<C: Context>(
     inactive: &HashSet<C::ValidatorId>,
     validator_stakes: BTreeMap<C::ValidatorId, U512>,
 ) -> Validators<C::ValidatorId> {
-    let sum_stakes: U512 = validator_stakes.values().copied().sum();
+    let sum_stakes = safe_sum(validator_stakes.values().copied()).expect("should not overflow");
```
Comment (alsrdn) [inline node/src/components/consensus/protocols/common.rs:20]: It would be also nice to have a unit test for this case also.

Comment (real-felix) [reply node/src/components/consensus/protocols/common.rs:20]: Where would you put it?

Comment (real-felix) [reply node/src/components/consensus/protocols/common.rs:20]: Done @alsrdn

**Review by alsrdn — APPROVED** [review]

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

**Pull request merged**